drivers/gpio/gpio_nrfx.c

```diff
@@ -192,40 +68,96 @@ static int gpio_nrfx_config(const struct device *port,
 		break;
 
 	default:
-		return -EINVAL;
+		drive = NRF_GPIO_PIN_D0H1;
+		break;
+	}
+
+	return drive;
+}
+
+static nrf_gpio_pin_pull_t get_pull(gpio_flags_t flags)
+{
+	if (flags & GPIO_PULL_UP) {
+		return NRF_GPIO_PIN_PULLUP;
+	} else if (flags & GPIO_PULL_DOWN) {
+		return NRF_GPIO_PIN_PULLDOWN;
 	}
 
-	if ((flags & GPIO_PULL_UP) != 0) {
-		pull = NRF_GPIO_PIN_PULLUP;
-	} else if ((flags & GPIO_PULL_DOWN) != 0) {
-		pull = NRF_GPIO_PIN_PULLDOWN;
-	} else {
-		pull = NRF_GPIO_PIN_NOPULL;
+	return NRF_GPIO_PIN_NOPULL;
+}
+
+static int pin_uninit(nrfx_gpiote_pin_t pin)
+{
+	uint8_t ch;
+	nrfx_err_t err;
+
+	err = nrfx_gpiote_pin_uninit(pin);
+	if (err != NRFX_SUCCESS) {
+		return -EIO;
+	}
+
+	nrf_gpio_cfg_default(pin);
+	err = nrfx_gpiote_channel_get(pin, &ch);
```

Comment:
Shouldn't this be called before `nrfx_gpiote_pin_uninit(pin)`? I think you'll always get `NRFX_ERROR_INVALID_PARAM` here after the pin is deinitialized.